apps/examples/testcase/ta_tc/ble_manager/itc/itc_blemanager_main.c

```diff
@@ -0,0 +1,865 @@
+
+	ble_client_ctx *ctx;
+	ctx = ble_client_create_ctx(&client_config);
+	ble_conn_info conn_info = { 0, };
+	memcpy(conn_info.addr.mac, g_target.mac, BLE_BD_ADDR_MAX_LEN);
+	conn_info.addr.type = BLE_ADV_TYPE_IND;
+	conn_info.conn_interval = 8;
+	conn_info.slave_latency = 3;
+	conn_info.mtu = 240;
+	conn_info.scan_timeout = 1000;
+	conn_info.is_secured_connect = true;
+
+	for (i = 1; i <= LOOP_SIZE; i++) {
+		gettimeofday(&start_t, NULL);
+		ret = ble_client_connect(ctx, &conn_info);
+		TC_ASSERT_EQ_CLEANUP("ble_client_connect", ret, BLE_MANAGER_SUCCESS, ble_manager_deinit());
+		ITC_FUNC_WAIT;
+		gettimeofday(&end_t, NULL);
+		average += diff_time(&start_t, &end_t);
+
+		ret = ble_client_disconnect(ctx);
+		TC_ASSERT_EQ_CLEANUP("ble_client_disconnect", ret, BLE_MANAGER_SUCCESS, ble_manager_deinit());
+		ITC_FUNC_WAIT;
+		sleep(1);
+	}
+
+	ret = ble_manager_deinit();
+	TC_ASSERT_EQ("ble_manager deinit", ret, BLE_MANAGER_SUCCESS);
+	printf("\n[Average connect time: %.2f ms]\n", average / (LOOP_SIZE * 1000));
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @testcase         itc_blemanager_average_disconnect_time
+ * @brief            check average time taken to disconnect from a server
+ * @scenario         check initialize, connect, disconnect,and de-initalization of ble manager
+ * @apicovered       ble_manager_init, ble_client_connect, ble_client_disconnect, ble_manager_deinit
+ * @precondition     none
+ * @postcondition    none
+ */
+static void itc_blemanager_average_disconnect_time(void)
+{
+	int i = 0;
+	double average = 0.0f;
+	struct timeval start_t;
+	struct timeval end_t;
+
+	ble_result_e ret = BLE_MANAGER_FAIL;
+	ret = ble_manager_init(NULL);
+	TC_ASSERT_EQ("ble_manager_init", ret, BLE_MANAGER_SUCCESS);
+
+	ble_client_ctx *ctx;
+	ctx = ble_client_create_ctx(&client_config);
+	ble_conn_info conn_info = { 0, };
+	memcpy(conn_info.addr.mac, g_target.mac, BLE_BD_ADDR_MAX_LEN);
+	conn_info.addr.type = BLE_ADV_TYPE_IND;
+	conn_info.conn_interval = 8;
+	conn_info.slave_latency = 3;
+	conn_info.mtu = 240;
+	conn_info.scan_timeout = 1000;
+	conn_info.is_secured_connect = true;
+
+	for (i = 1; i <= LOOP_SIZE; i++) {
+		ret = ble_client_connect(ctx, &conn_info);
+		TC_ASSERT_EQ_CLEANUP("ble_client_connect", ret, BLE_MANAGER_SUCCESS, ble_manager_deinit());
+		ITC_FUNC_WAIT;
+
+		gettimeofday(&start_t, NULL);
+		ret = ble_client_disconnect(ctx);
+		TC_ASSERT_EQ_CLEANUP("ble_client_disconnect", ret, BLE_MANAGER_SUCCESS, ble_manager_deinit());
+		ITC_FUNC_WAIT;
+		gettimeofday(&end_t, NULL);
+		average += diff_time(&start_t, &end_t);
+	}
+
+	ret = ble_manager_deinit();
+	TC_ASSERT_EQ("ble_manager deinit", ret, BLE_MANAGER_SUCCESS);
+	printf("\n[Average disconnect time: %.2f ms]\n", average / (LOOP_SIZE * 1000));
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @testcase         itc_blemanager_average_stoping_time
+ * @brief            check average time taken to deinit
+ * @scenario         check initialize, de-initalization of ble manager
```

Comment:
As @abhishek-samsung mentioned, kindly consider using initialization instead of initialize in descriptions of all the functions.

Reply:
Sure, done